src/main/webapp/configure.js

```diff
@@ -153,6 +158,89 @@ async function loadBrowserLabInfo(a, b, o, path, err) {
     }
 }
 
+async function loadWorkspaceInfo(a, b, o, err) {
+    let baseUrl = "";
+    await a.getMcServerUrl(o.serverName, async (r) => {
+        baseUrl = r.responseObject();
+        if (baseUrl) {
+            baseUrl = baseUrl.trim().replace(/[\/]+$/, "");
+        } else {
+            err.style.display = "inline-block";
+            await triggerBtnState(b, false);
+            return;
+        }
+
+        await a.getValidWorkspaces(baseUrl,o.authType, o.userName, o.password, o.execToken, o.useProxy, o.proxyAddress, o.useProxyAuth, o.proxyUserName, o.proxyPassword, async (response) => {
+            try {
+                let responseData = response.responseObject();
+
+                if (Array.isArray(responseData)) {
+
+                    // Try to find workspace-like objects (looking for name/uuid properties)
+                    const workspaceObjects = responseData.filter(item =>
+                        item && (item.name || item.uuid)
+                    );
+
+                    if (workspaceObjects.length > 0) {
+                        const div = o.workspaceInfo;
+                        const selectElement = div.querySelector('select[name="workspaceSelect"]');
+                        const previousSelectedWorkspace = selectElement.value;
+                        const saveWorkspace = renderWorkspaces(workspaceObjects, selectElement);
+                        selectElement.innerHTML = saveWorkspace;
+
+                        const workspaceIdElement = div.querySelector('input[name="workspaceId"]');
+                        const workspaceNameElement = div.querySelector('input[name="workspaceName"]');
+
+                        if (workspaceObjects.some(workspace => workspace.uuid === previousSelectedWorkspace)) {
+                            const selectedWorkspace = workspaceObjects.find(workspace => workspace.uuid === previousSelectedWorkspace);
+                            workspaceIdElement.value = selectedWorkspace.uuid;
+                            workspaceNameElement.value = selectedWorkspace.name;
+                            selectElement.value = previousSelectedWorkspace;
+                        } else {
+                            // Use the first item in the list if the previous selected item is not found
+                            ({ uuid: workspaceIdElement.value, name: workspaceNameElement.value } = workspaceObjects[0]);
+                        }
```

Comment:
Please try this simplified alternative (no "if"/"else" needed):
const selectedWorkspace = workspaceObjects.find(workspace => workspace.uuid === previousSelectedWorkspace) ?? workspaceObjects[0];
workspaceIdElement.value = selectedWorkspace.uuid;
workspaceNameElement.value = selectedWorkspace.name;
selectElement.value = selectedWorkspace.uuid;